### PR TITLE
Add a blog post about GSoC & Chrome Extensions

### PR DESCRIPTION
**Pull request by daidr**

### CODE REVIEW

File: site/_data/i18n/authors.yaml [hunk 1]
```diff
@@ -1531,6 +1531,12 @@ samrichard:
     en: 'Sam Richard'
   description:
     en: 'Developer Advocate, Google ChromeOS'
+xuezhoudai:
+  title:
+    en: 'Xuezhou Dai'
+    zh: '戴学舟'
+  description:
+    en: 'A contributor to Chrome Developers'
```
Comment (oliverdunk) [inline site/_data/i18n/authors.yaml:1539]: Maybe something like "Summer of Code '23 Contributor"?

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+* I'm introverted, and English is not my native language, so communication with mentors might be challenging.
+* GSoC communities come from all over the world, and dealing with time zone differences could be difficult.
+* GSoC is such a renowned program that it can be competitive, making my chances seem slim.
+
+However, I can confidently say none of these were an issue!
+
+Before I tell my story, I'd like to talk about something that happened to me at the beginning of 2022. This incident was also an opportunity for me to participate in GSoC. I had wanted to migrate a browser extension based on Manifest V2 to Manifest V3. In order to complete the migration, I had to refer to the documentation and understand every API change. I also had to search for whether there were relevant samples for reference. Comprehending the new API and transferring the code is quite challenging for me.
+
+That's why I was highly excited when browsing the GSoC project list and catching the idea related to mv3. I'm eager to contribute to improving the Manifest V3 extension samples and provide an amenity for future developers.
```
Comment (oliverdunk) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:39]:
```suggestion
That's why I was highly excited when browsing the GSoC project list and catching the idea related to MV3. I'm eager to contribute to improving the Manifest V3 extension samples and provide an amenity for future developers.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+In March, I emailed Oliver, one of the mentors in charge, a brief email about my intention:
+
+{% Details %}
+{% DetailsSummary %}
+The application for participate in GSoC project
+{% endDetailsSummary %}
+
+Hi,
+
+I am a CS undergraduate student from China. I learned about the Chrome Extensions Samples related project at GSoC and I was interested in improving the Manifest V3 Samples. I don't have much prior experience in working on API Samples, however, I have experience in web development and mv3 extension development (https://github.com/daidr/paimon-webext), which has made me realize the importance of mv3 api samples, so I would like both to learn and to contribute to it. Can I still apply for this project?
```
Comment (oliverdunk) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:50]:
```suggestion
I am a CS undergraduate student from China. I learned about the Chrome Extensions Samples related project at GSoC and I was interested in improving the Manifest V3 Samples. I don't have much prior experience in working on API Samples, however, I have experience in web development and MV3 extension development (https://github.com/daidr/paimon-webext), which has made me realize the importance of MV3 API samples, so I would like both to learn and to contribute to it. Can I still apply for this project?
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+This summer has given me many indelible memories. Coding makes me highly content. I have encountered patient mentors who guided me and taught me interesting knowledge, enriching my experience.
+
+The knowledge in the field of computer science is endless. When I compared myself with others, I realized my understanding of open source was only the tip of the iceberg. However, every little bit helps. I believe GSoC is a great starting point for participating in open source.
+
+I also hope my experience can provide some reference for future participation in open-source activities.
+
+Here are some of my suggestions anyone wanting to participate in Google Summer of Code:
```
Comment (oliverdunk) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:95]:
```suggestion
Here are some of my suggestions for anyone wanting to participate in Google Summer of Code:
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
+---
+layout: 'layouts/blog-post.njk'
+title: Google Summer of Code & Chrome Extensions
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:3]:
```suggestion
title: Google Summer of Code and Chrome Extensions
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+description: >
+  The experience and advice on participating in Google Summer of Code.
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:5]:
```suggestion
  Experience and advice on participating in Google Summer of Code.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+date: 2023-08-29
+authors:
+  - xuezhoudai
+
+hero: 'image/wVNVUJS8Z8O04i1tJKSdsp6nkRQ2/8ZNtuWAIcc7NBdS3AQEv.jpg'
+alt: >
+  Google Summer of Code & Chrome Extensions
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:12]:
```suggestion
  Google Summer of Code and Chrome Extensions
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+sharing_image: 'image/wVNVUJS8Z8O04i1tJKSdsp6nkRQ2/8ZNtuWAIcc7NBdS3AQEv.jpg'
+---
+
+I'm a sophomore from China passionate about web development. In my first year, I joined a technical club at our college. This club was my introduction to coding and open source. In the club, I met a group of like-minded partners who love coding. And it was from them that I learned about Google Summer of Code at the beginning of this year. This Google-organized global program connects students with open-source organizations and guides them in making good use of their summer by participating in open-source activities.
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:17]:
```suggestion
I'm a sophomore from China passionate about web development. In my first year, I joined a technical club at our college. This club was my introduction to coding and open source. In the club, I met a group of like-minded partners who love coding. And it was from them that I learned about Google Summer of Code at the beginning of 2023. This Google-organized global program connects students with open-source organizations and guides them in making good use of their summer by participating in open-source activities.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+With the attitude of having a try, I decided to submit my application. Fortunately, I was accepted. The summer I spent contributing to the Chrome Extension Samples repo has been memorable and precious. Of course, I have also learned a lot: communicating effectively, coding skills, and planning abilities, among other things.
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:19]:
```suggestion
Thinking, "why not?", I submitted my application. Fortunately, I was accepted. The summer I spent contributing to the Chrome Extension Samples repo has been memorable and precious. Of course, I've also learned a lot: effective communication, coding skills, and planning abilities, among other things.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+As GSoC 2023 is drawing to a close, it's worth sharing some of my experiences in GSoC. This post will briefly introduce the general process of GSoC through my own participation, hoping to be helpful to you!
+
+## The story of how I contributed to GSoC
+
+I applied for the Chromium GSoC 2023 Project, where my main task would be upgrading existing Chrome extension samples to work in Manifest V3, including some related scripts and documents.
+
+### At the Very Beginning
+
+I first learned of GSoC in February 2023 but was not yet  sure I wanted to apply. I had a few concerns back then:
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:29]:
```suggestion
I first learned of GSoC in February 2023 but was not yet sure I wanted to apply. I had a few concerns back then:
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+* I'm introverted, and English is not my native language, so communication with mentors might be challenging.
+* GSoC communities come from all over the world, and dealing with time zone differences could be difficult.
+* GSoC is such a renowned program that it can be competitive, making my chances seem slim.
+
+However, I can confidently say none of these were an issue!
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:35]:
```suggestion
However, I can confidently say none of these were an issue.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+Before I tell my story, I'd like to talk about something that happened to me at the beginning of 2022. This incident was also an opportunity for me to participate in GSoC. I had wanted to migrate a browser extension based on Manifest V2 to Manifest V3. In order to complete the migration, I had to refer to the documentation and understand every API change. I also had to search for whether there were relevant samples for reference. Comprehending the new API and transferring the code is quite challenging for me.
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:37]:
```suggestion
Before I tell my story, I'd like to talk about something that happened to me at the beginning of 2022. This incident was also an opportunity for me to participate in GSoC. I had wanted to migrate a browser extension based on Manifest V2 to Manifest V3. To complete the migration, I had to refer to the documentation and understand every API change. I also had to search for whether there were relevant samples for reference. Comprehending the new API and transferring the code is quite challenging for me.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+
+That's why I was highly excited when browsing the GSoC project list and catching the idea related to MV3. I'm eager to contribute to improving the Manifest V3 extension samples and provide an amenity for future developers.
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:39]:
```suggestion
That's why I was highly excited when browsing the GSoC project list and catching the idea related to MV3. I was eager to contribute to improving the Manifest V3 extension samples and provide an amenity for future developers.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+
+In March, I emailed Oliver, one of the mentors in charge, a brief email about my intention:
+
+{% Details %}
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:43]:
```suggestion
{% Details 'open' %}
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+{% DetailsSummary %}
+The application for participate in GSoC project
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:45]:
```suggestion
The application to participate in the GSoC project
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+
+For this larger task, we decided that we needed an automated script in the samples repo to traverse the repository and generate a JSON file containing a list of extension samples and the APIs they use. The developer.chrome.com documentation repo now uses this file to build a page.
+
+I first considered using Babel to analyze the sample code's [Abstract Syntax Tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) and identify its used APIs. After sharing this idea with my mentor, I discovered they had previously worked on something similar, confirming this approach's feasibility. After I completed the code, my mentor and other members of his team provided valuable suggestions including paying attention to code segmentation, standardizing code style, and documenting functions properly.
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:79]:
```suggestion
I first considered using Babel to analyze the sample code's [Abstract Syntax Tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) and identify the APIs it used. After sharing this idea with my mentor, I discovered they had previously worked on something similar, confirming this approach's feasibility. After I completed the code, my mentor and other members of his team provided valuable suggestions including paying attention to code segmentation, standardizing code style, and documenting functions properly.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+
+After several big and small reviews, the code was successfully merged. You can find the whole thread on [GitHub](https://github.com/GoogleChrome/developer.chrome.com/pull/7029), and the live page is available [here](/docs/extensions/samples).
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:81]:
```suggestion
After several big and small reviews, the code was successfully merged. You can find the whole thread on [GitHub](https://github.com/GoogleChrome/developer.chrome.com/pull/7029), or [view the live page](/docs/extensions/samples).
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+
+## Wrapping Up
+
+If you ask me what GSoC has brought me: first of all, of course, is understanding the knowledge related to browser extension APIs. By reading the documentation and writing new samples, I gained a detailed understanding of various browser extension APIs.
+
+Additionally, I have acquired many skills I had little exposure to before being accepted for GSoC, such as following a comprehensive code review process and creating PRDs. I have also started using GitHub actions to automate workflows and learned how to use Babel to traverse all extension samples and track the APIs used. Moreover, I had my first experience with the Nunjucks template engine. These abilities are essential for programming, and I am delighted that I learned them in advance. Participating in open-source communities brings me joy.
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:87]:
```suggestion
Additionally, I have acquired many skills I had little exposure to before being accepted for GSoC, such as following a comprehensive code review process and creating a [product requirements doc](https://en.wikipedia.org/wiki/Product_requirements_document) (PRD). I've also started using GitHub actions to automate workflows and learned how to use Babel to traverse all extension samples and track the APIs used. Moreover, I had my first experience with the Nunjucks template engine. These abilities are essential for programming, and I am delighted that I learned them. Participating in open-source communities brings me joy.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+
+I also hope my experience can provide some reference for future participation in open-source activities.
+
+Here are some of my suggestions for anyone wanting to participate in Google Summer of Code:
+
+1. Choose an idea that suits you: In my opinion, when looking for ideas, interest should be your priority, followed by skills. This strategy will allow you to maintain enthusiasm throughout the contribution process.
+2. Quality over quantity: You can submit many proposals to increase the chances of being accepted for GSoC. However, I believe that quality is probably more critical because human energy is limited. Focusing on two or three ideas may be more helpful in the end.
+3. Don't be afraid of new things: Don't hesitate to try new technologies you haven't experienced before in a project. You have nearly a month of community bonding to familiarize yourself with unfamiliar technologies and a whole summer to implement them. Just go for it!
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:99]:
```suggestion
3. Don't be afraid of new things: Don't hesitate to try technologies you haven't worked with in a project. You have nearly a month of community bonding to familiarize yourself with unfamiliar technologies and a whole summer to implement them. Just go for it!
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+4. Get acquainted with the project in advance: For example, read through development documentation, look at previous commit history and issues, and skim through past pull requests. Getting an overall understanding of the project in advance will help you write a more comprehensive proposal.
+5. Maintain communication: Regularly communicate with your mentor, and don't hesitate to ask questions when confronting problems, whether before applying or after being accepted. Most mentors are willing to help you solve problems; this way, they can also keep track of your progress.
+
+Thank you for reading!
```
Comment (jpmedley) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:103]:
```suggestion
Thank you for reading.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]
Please don't address any of my changes yet. Given our discussion the other day, I'd like @AmySteam to take a pass and I want her to see my suggestions.

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md (hunk 9 above)
Comment (AmySteam) [reply site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:19]: I think the first part is more like "giving it a try". IMO The rest seems okay to leave the original text.

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md (hunk 20 above)
Comment (AmySteam) [reply site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:103]: I would leave the "!"... seems to flow better with the tone of the blog :)

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md (hunk 9 above)
Comment (jpmedley) [reply site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:19]: Done.

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md (hunk 20 above)
Comment (jpmedley) [reply site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:103]: I want to keep this comment. This is actually an incorrect use of the exclamation point.

**Comment by jpmedley** [comment]
@oliverdunk Amy's review is done. You have the ball.

**Comment by oliverdunk** [comment]
> @oliverdunk Amy's review is done. You have the ball.

Thanks! Will ask @daidr to take a look.

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+
+After several big and small reviews, the code was successfully merged. You can find the whole thread on [GitHub](https://github.com/GoogleChrome/developer.chrome.com/pull/7029), or [view the live page](/docs/extensions/samples).
+
+## Wrapping Up
+
+If you ask me what GSoC has brought me: first of all, of course, is understanding the knowledge related to browser extension APIs. By reading the documentation and writing new samples, I gained a detailed understanding of various browser extension APIs.
+
+Additionally, I have acquired many skills I had little exposure to before being accepted for GSoC, such as following a comprehensive code review process and creating a [product requirements doc](https://en.wikipedia.org/wiki/Product_requirements_document) (PRD). I've also started using GitHub actions to automate workflows and learned how to use Babel to traverse all extension samples and track the APIs used. Moreover, I had my first experience with the Nunjucks template engine. These abilities are essential for programming, and I am delighted that I learned them. Participating in open-source communities brings me joy.
```
Comment (oliverdunk) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:87]:
```suggestion
Additionally, I have acquired many skills I had little exposure to before being accepted for GSoC, such as following a comprehensive code review process and creating a PRD. I've also started using GitHub actions to automate workflows and learned how to use Babel to traverse all extension samples and track the APIs used. Moreover, I had my first experience with the Nunjucks template engine. These abilities are essential for programming, and I am delighted that I learned them. Participating in open-source communities brings me joy.
```

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+Writing a proposal is the last thing to do before applying. The proposal should include ideas, goals, and a schedule in addition to the application form. Depending on the
+project you're hoping to work on, you may be provided with an additional template for the proposal. Many ready-made proposal examples are available online for reference.
+
+After completing the draft, I sent it to my mentor for feedback. Once the application is submitted, there's a long wait. And in early May, I received a notification that I was accepted.
+
+### Code Work
+
+During the initial bonding period, I first dealt with the remaining starter bugs and checked if there were any new skills I had to learn. And I spend the rest of the time reading documentation.
+
+After a discussion with my mentor, we adjusted some goals to make the time allocated to tasks more reasonable. We also discussed an idea for creating a new developer.chrome.com page listing all API samples and providing basic filters to help developers locate the needed samples quickly. It was a rather complex task for me, and it took my mentor and me quite some time to define the project and write a product requirements document.
```
Comment (oliverdunk) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:75]:
```suggestion
After a discussion with my mentor, we adjusted some goals to make the time allocated to tasks more reasonable. We also discussed an idea for creating a new developer.chrome.com page listing all API samples and providing basic filters to help developers locate the needed samples quickly. It was a rather complex task for me, and it took my mentor and me quite some time to define the project and write a [product requirements doc](https://en.wikipedia.org/wiki/Product_requirements_document) (PRD).
```

**Review by oliverdunk — CHANGES_REQUESTED** [review]
Left two suggestions (I think we should define PRD the first time we say it, rather than the second). Otherwise this looks good!

**Review by oliverdunk — APPROVED** [review]
LGTM 🚀

**Review by jpmedley — APPROVED** [review]

**Comment by jpmedley** [comment]
@oliverdunk What day is this scheduled?

File: site/en/blog/google-summer-of-code-and-chrome-extensions/index.md [hunk 23]
```diff
@@ -0,0 +1,103 @@
+---
+layout: 'layouts/blog-post.njk'
+title: Google Summer of Code and Chrome Extensions
+description: >
+  Experience and advice on participating in Google Summer of Code.
+date: 2023-08-29
```
Comment (oliverdunk) [inline site/en/blog/google-summer-of-code-and-chrome-extensions/index.md:6]:
```suggestion
date: 2023-09-04
```

**Pull request merged**